servicetalk-tcp-netty-internal/src/main/java/io/servicetalk/tcp/netty/internal/TcpServerConfig.java

```diff
@@ -87,6 +118,29 @@ public void transportObserver(final TransportObserver transportObserver) {
         this.transportObserver = requireNonNull(transportObserver);
     }
 
+    /**
+     * Add SSL/TLS related config.
+     *
+     * @param sslConfig the {@link ServerSslConfig}.
+     */
+    public void sslConfig(final @Nullable ServerSslConfig sslConfig) {
+        this.sslConfig = sslConfig;
+        this.acceptInsecureConnections = false;
+    }
+
+    /**
+     * Set the SSL/TLS configuration and allows to specify if insecure connections should also be allowed.
+     *
+     * @param config The configuration to use.
+     * @param acceptInsecureConnections if non-TLS connections are accepted on the same socket.
+     * @return {@code this}.
+     */
+    public TcpServerConfig sslConfig(ServerSslConfig config, boolean acceptInsecureConnections) {
+        sslConfig(requireNonNull(config));
+        this.acceptInsecureConnections = acceptInsecureConnections;
```

Comment:
I think we still have a few problems with these overloads.
1. This is the only signature enforcing that `ServerSslConfig != null` but I think all but the single param version expect it to be non null.
2. We're still keeping a false memory of the `maxClientHelloLength` and `clientHelloTimeout` settings, although this is not your doing.

Maybe we should funnel everything through a `private setSslConfig(*all params*)` method and make each overload to defer to it and fill in missing params with defaults. Since this predates this PR (although the proliferation makes it worse) there is a good argument to do it as a follow up.

Reply:
👍 I will do this as a follow up, thanks!